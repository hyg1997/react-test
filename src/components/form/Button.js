import styled from "styled-components";
import {Button as AntButton} from "antd";
import {get} from "lodash";

export const Button = styled(AntButton)`
  text-transform: uppercase;
  margin-right: 1rem;
  width: ${props => get(props, "width", "100px")};
  height: 50px;
`;
