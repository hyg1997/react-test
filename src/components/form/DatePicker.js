import React, {forwardRef, Fragment} from "react";
import styled from "styled-components";
import {colors} from "../../constants/colors";
import {sizes} from "../../constants/sizes";
import {DatePicker as AntDatePicker} from "antd";
import {useI18n} from "../../utils";

export const DatePicker = forwardRef((props, ref) => {
    const i18n = useI18n();

    return (
        <Fragment>
            {
                props.label &&
                <Label required={props.required}>{props.label}</Label>
            }
            <DatePickerContainer>
                <StyledDatePicker {...props}
                                  placeholder={i18n(["general", "selectDate"])}
                                  hasError={props.error}
                                  ref={ref}
                />
            </DatePickerContainer>
            {
                props.error &&
                <Error>{props.errorMessage || props.error.message}</Error>
            }
        </Fragment>
    )
});

const DatePickerContainer = styled.div`
  position: relative;
  display: flex;
  flex-direction: row;
  align-items: center;
`;

const StyledDatePicker = styled(AntDatePicker)`
  margin-bottom: 1rem !important;
  ${props => props.hasError && `
    background-color: #fff;
    border-color: ${colors.danger};
  `}
`;

const Label = styled.label`
  display: block;
  margin-bottom: 0.5rem;
  font-size: ${sizes.font.normal};
  
  ${props => props.required && `
    ::before {
        display: inline-block;
        margin-right: 4px;
        color: #f5222d;
        font-size: 14px;
        line-height: 1;
        content: "*";
    }
  `}
`;

const Error = styled.p`
  margin-top: -0.5rem;
  font-size: ${sizes.font.small};
  color: ${colors.danger};
`;
