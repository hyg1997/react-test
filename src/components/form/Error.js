import styled from "styled-components";
import {sizes} from "../../constants/sizes";
import {colors} from "../../constants/colors";

export const Error = styled.p`
  font-size: ${sizes.font.small};
  color: ${colors.danger};
`;
