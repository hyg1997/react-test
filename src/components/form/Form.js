import React, {useEffect} from "react";
import {useGlobal} from "reactn";

export const Form = props => {
    const [globalCurrentLanguageCode] = useGlobal("currentLanguageCode");

    useEffect(() => {
        const {clearError, triggerValidation} = props;

        const updateErrors = async () => {
            await triggerValidation();
            clearError();
        };

        updateErrors();
    }, [globalCurrentLanguageCode]);

    return (
        <form {...props} noValidate>
            {props.children}
        </form>
    );
};
