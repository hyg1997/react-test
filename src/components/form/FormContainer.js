import styled from "styled-components";
import {responsive} from "../../constants/responsive";

export const FormContainer = styled.div`
  width: 280px;

  @media(min-width: ${responsive.tablet}) {
      width: 500px;
  }
`;
