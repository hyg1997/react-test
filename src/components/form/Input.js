import React, {forwardRef, Fragment, useEffect, useState} from "react";
import {Icon} from "../Icon";
import {faEye, faEyeSlash} from "@fortawesome/free-solid-svg-icons"
import styled from "styled-components";
import {colors} from "../../constants/colors";
import {sizes} from "../../constants/sizes";

export const Input = forwardRef((props, ref) => {
    const [hide, setHide] = useState(false);

    const inputType = () => {
        if (props.type === "password") {
            return hide ? "password" : "text";
        }
        return props.type;
    };

    useEffect(() => {
        if (props.type === "password") setHide(true);
    }, [props.type]);

    return (
        <Fragment>
            {props.label && <Label required={props.required}>{props.label}</Label>}
            <InputContainer>
                <StyledInput {...props} hasError={props.error}
                             ref={ref}
                             type={inputType()}/>
                {props.type === "password" &&
                <StyledIcon icon={hide ? faEye : faEyeSlash}
                            onClick={() => setHide(!hide)}/>}
            </InputContainer>
            {props.error && <Error>{props.errorMessage || props.error.message}</Error>}
        </Fragment>
    )
});


const InputContainer = styled.div`
  position: relative;
  display: flex;
  flex-direction: row;
  align-items: center;
`;

const StyledIcon = styled(Icon)`
  position: absolute;
  right: 0;
  bottom: 50%;
`;

const StyledInput = styled.input`
  margin-bottom: 1rem !important;
  ${props => props.hasError && `
    background-color: #fff;
    border-color: ${colors.danger};
  `}
`;

const Label = styled.label`
  display: block;
  margin-bottom: 0.5rem;
  font-size: ${sizes.font.normal};
  
  ${props => props.required && `
    ::before {
        display: inline-block;
        margin-right: 4px;
        color: #f5222d;
        font-size: 14px;
        line-height: 1;
        content: "*";
    }
  `}
`;

const Error = styled.p`
  margin-top: -0.5rem;
  font-size: ${sizes.font.small};
  color: ${colors.danger};
`;
