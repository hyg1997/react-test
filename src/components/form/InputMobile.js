import React, {forwardRef, Fragment} from "react";
import styled from "styled-components";
import {colors} from "../../constants/colors";
import {sizes} from "../../constants/sizes";
import {InputItem as AntInputItem} from "antd-mobile";

export const InputItem = forwardRef((props, ref) => {
    return (
        <Fragment>
            {
                props.label &&
                <Label required={props.required}>{props.label}</Label>
            }
            <StyledInputItem {...props}
                             hasError={props.error}
                             ref={ref}/>
            {
                props.error &&
                <Error>{props.errorMessage || props.error.message}</Error>
            }
        </Fragment>
    )
});

const StyledInputItem = styled(AntInputItem)`
  margin-bottom: 1rem !important;
  ${props => props.hasError && `
    background-color: #fff;
    border-color: ${colors.danger};
  `}
`;

const Label = styled.label`
  display: block;
  margin-bottom: 0.5rem;
  font-size: ${sizes.font.normal};
  
  ${props => props.required && `
    ::before {
        display: inline-block;
        margin-right: 4px;
        color: #f5222d;
        font-size: 14px;
        line-height: 1;
        content: "*";
    }
  `}
`;

const Error = styled.p`
  margin-top: -0.5rem;
  font-size: ${sizes.font.small};
  color: ${colors.danger};
`;
