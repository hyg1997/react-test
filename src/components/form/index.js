export * from "./Input";
export * from "./Button";
export * from "./Select";
export * from "./Upload";
export * from "./FormContainer";
export * from "./Error";
export * from "./DatePicker";
export * from "./Radio";
export * from "./Checkbox";
export * from "./Switch";
export * from "./Form";
export * from "./InputMobile";
