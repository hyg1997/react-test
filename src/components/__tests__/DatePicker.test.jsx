import React from "react";
import { shallow, configure } from "enzyme";
import { DatePicker } from "../form";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("DatePicker", () => {
  it("Renders correctly", () => {
    const wrapper = shallow(<DatePicker label="Date" required />);
    expect(wrapper).toMatchSnapshot();
  });
});
