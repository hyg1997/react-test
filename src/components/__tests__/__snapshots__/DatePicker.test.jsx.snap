// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker Renders correctly 1`] = `ShallowWrapper {}`;
